<hr>
<b><font color='red'>Ejercicio 01</font></b>  

Lee el archivo.txt 

In [1]:
f = open("archivo.txt", "r").readlines()

for string in f:
    a = string.split()
    for i in range(0,len(a),2):
        print("%.1f" % int(a[i]) , "%.2f" % int(a[i+1]))

1.0 2.00
2.0 4.00
3.0 6.00
4.0 8.00


y haz que tenga la siguiente salida:

    1.0   2.00  
    2.0   4.00  
    3.0   6.00  
    4.0   8.00  


<hr>
<b><font color='red'>Ejercicio 02</font></b>  

Haz lo mismo con el archvo archivo_error.txt,con ciddado de que a este archivo le falta algún dato, haz que en ese caso cuando lo lea, si no hay dos columnas no lo escriba.

In [2]:
"""Esribe tu código aquí"""
archivo="archivo_error.txt"
texto=""
for line in open(archivo).readlines():
    if(len(line.split()) >= 2):
        texto=texto+line
print(texto)

1 2
3 6
4 8



<hr>
<b><font color='red'>Ejercicio 03</font></b>  

Lee el archivo.txt, haz que imprima las columnas en las filas y las filas en las columnas

In [3]:
archivo="archivo.txt"
text=open(archivo).readlines()
T=[]
for line in text:
    a=[]
    for i in line.split():
        a.append(i)
    T.append(a)
#print(T)
for i in range(len(T[0])):
    a=""
    for j in T:
        a=a+j[i]+" "
    print(a)

1 2 3 4 
2 4 6 8 


<hr>
<b><font color='red'>Ejercicio 04</font></b>  

1. Escribe en un archivo la siguiente matriz:
$
\begin{equation}
\begin{pmatrix}
3 & 1 & 3\\
-1 & 1 & 0\\
-2 & 4 & 1
\end{pmatrix}
\end{equation} 
$
2. Lee el archivo
3. Calcula con los datos leidos el determinante de la matriz y lo saque por pantalla

In [4]:
f = open("matriz.txt", "w")
f.write('3 1 3\n')
f.write('-1 1 0\n')
f.write('-2 4 1')
f.close()
f = open("matriz.txt", "r").readlines()
m = []
for line in f:
    aux=[]
    for i in line.split():
        aux.append(float(i))
    m.append(aux)
print("Resultado de la matriz: ", m)

i =(  m[0][0]*m[1][1]*m[2][2]
    + m[1][0]*m[2][1]*m[0][2]
    + m[2][0]*m[0][1]*m[1][2]
    - m[2][0]*m[1][1]*m[0][2]
    - m[0][0]*m[2][1]*m[1][2]
    - m[1][0]*m[0][1]*m[2][2])
    
print(i)

Resultado de la matriz:  [[3.0, 1.0, 3.0], [-1.0, 1.0, 0.0], [-2.0, 4.0, 1.0]]
-2.0


<hr>
<b><font color='red'>Ejercicio 05</font></b>  

En el criptoanálisis, la técnica de análisis de frecuencias consiste en el aprovechamiento de estudios sobre la frecuencia de las letras o grupos de letras en los idiomas para poder establecer hipótesis para aprovecharlas para poder descifrar un texto cifrado sin tener la clave de descifrado (romper). Es un método típico para romper cifrados clásicos. 
Haz una anailisis de frecuencias para los 10 primeros càpitulos de el libro "el_quijote.txt"

In [5]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
f=[]
for i in abc:
    f.append(1)
N=0
archivo="el_quijote.txt"
text=open(archivo).readlines()
nmaxlines=len(text)
for line in text:
    for word in line.split():
        for i in word.upper():
            for j in abc:
                a=i.upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U")
                if a==j :                 
                    f[(abc.index(i))]= f[(abc.index(i))]+1 
                    N=N+1
        
for i in abc:
    f[(abc.index(i))]=f[(abc.index(i))]/N
    
for i in abc:
   print("%.2c  " % i,"%.3f" % f[(abc.index(i))])

A   0.127
B   0.017
C   0.037
D   0.053
E   0.138
F   0.005
G   0.011
H   0.012
I   0.053
J   0.007
K   0.000
L   0.060
M   0.026
N   0.066
Ñ   0.000
O   0.099
P   0.021
Q   0.019
R   0.061
S   0.073
T   0.037
U   0.048
V   0.011
W   0.000
X   0.000
Y   0.015
Z   0.004


Salida:

    A   0.127
    B   0.017
    C   0.037
    D   0.053
    E   0.138
    F   0.005
    .
    .
    .

<hr>
<b><font color='red'>Ejercicio 06</font></b>  

haz un analisis de frecuencias en el libro encriptado "RH RQCEPESEFM AR MPM.txt", y comparala con el analisis anterior


In [6]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
f2=[]
for i in abc:
    f2.append(1)
N=0
archivo="RH RQCEPESEFM AR MPM.txt"
text=open(archivo).readlines()
nmaxlines=len(text)
for line in text:
    for word in line.split():
        for i in word.upper():
            for j in abc:
                a=i.upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U")
                if a==j :                 
                    f2[(abc.index(i))]=f2[(abc.index(i))]+1 
                    N=N+1
        
for i in abc:
    f2[(abc.index(i))]=f2[(abc.index(i))]/N
    
print("el_quijote.txt","  RH RQCEPESEFM AR MPM.txt")    
for i in abc:
    print("%.2c  " % i,"%.3f" % f[(abc.index(i))],"       %.2c  " % i,"%.3f" % f2[(abc.index(i))])

el_quijote.txt   RH RQCEPESEFM AR MPM.txt
A   0.127        A   0.049
B   0.017        B   0.006
C   0.037        C   0.040
D   0.053        D   0.061
E   0.138        E   0.126
F   0.005        F   0.007
G   0.011        G   0.000
H   0.012        H   0.052
I   0.053        I   0.032
J   0.007        J   0.012
K   0.000        K   0.068
L   0.060        L   0.001
M   0.026        M   0.093
N   0.066        N   0.027
Ñ   0.000        Ñ   0.011
O   0.099        O   0.010
P   0.021        P   0.066
Q   0.019        Q   0.069
R   0.061        R   0.136
S   0.073        S   0.017
T   0.037        T   0.044
U   0.048        U   0.045
V   0.011        V   0.009
W   0.000        W   0.000
X   0.000        X   0.002
Y   0.015        Y   0.009
Z   0.004        Z   0.004


salida:

    el_quijote.txt   RH RQCEPESEFM AR MPM.txt
    A   0.127        A   0.049
    B   0.017        B   0.006
    C   0.037        C   0.040
    .                .
    .                .
    .                .

<hr>
<b><font color='red'>Ejercicio 07</font></b>  

Comparando el analisis de frecuencia sabrías decir cual es el titulo del libro.  
RH RQCEPESEFM AR MPM

<hr>
<b><font color='red'>Generar el ejercicio "El escarabajo de oro"</font></b>

En este cuento de Edgar Allan Poe, el protagonista (el señor Legrand) encuentra un tesoro gracias a que es capaz de descifrar un texto en clave con las pistas suficientes para ir en su busca.  

El señor Legrand consigue descifrar el texto cifrado porque va sustituyendo cada signo (cifra o símbolo) por una letra del abecedario, teniendo en cuenta que los signos que más se repiten deben corresponder, más o menos, a las letras que más se usan en español: la E, la A, la O... Después del mensaje en clave puedes ver, en rojo, el abecedario ordenado por orden de frecuencia: la letra más usada es la E, seguida de la A, etc.

In [7]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
rem="ESCARBJODFGHIKLMNÑPQTUVWXYZ"
f=[]
for i in abc:
    f.append(1)

texto=""
archivo="El_escarabajo_de_oro_SOL.txt"
text=open(archivo).readlines()
nmaxlines=len(text)
for line in text:
    for word in line.split():
        for i in word.upper():
            sust=False
            for j in abc:
                a=i.upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U")
                if a==j :                 
                    texto=texto+rem[abc.index(j)]
                    sust=True
            if not sust:
                texto=texto+i 
        texto=texto+' '
    texto=texto+'\n'
        
print(texto)

RH RQCEPESEFM AR MPM 

OECR IUCOMQ ELMQ TPESR DKTDIE EIDQTEA CMK UK CESEHHRPM HHEIEAM WDHHDEI HRJPEKA. ARQCRKADE AR UKE EKTDJUE BEIDHDE NPMTRQTEKTR Y RK UK TDRINM OESDE ADQBPUTEAM AR JPEK BMPTUKE, OEQTE ÑUR UKE QRPDR AR ARQJPECDEQ HM PRAUFRPMK E HE NMSPRZE. NEPE RVDTEP RH SMCOMPKM ÑUR QDJUR E TEHRQ ARQEQTPRQ, ESEKAMKM KURVE MPHREKQ, HE CDUAEA AR QUQ ESURHMQ, Y QR DKQTEHM RK HE DQHE AR QUHHDVEK, CRPCE AR COEPHRQTMK, RK HE CEPMHDKE ARH QUP. 
RQTE DQHE RQ IUY CUPDMQE. HE BMPIE CEQD NMP CMINHRTM HE EPRKE ARH IEP Y TDRKR UKEQ TPRQ IDHHEQ AR HEPJM. QU EKCOM KM RXCRAR RK KDKJUK NUKTM AR UK CUEPTM AR IDHHE. QR RKCURKTPE QRNEPEAE AR TDRPPE BDPIR NMP UK EPPMYM ENRKEQ NRPCRNTDSHR, ÑUR QR DKQDKUE RK UKE ARQMHEAE ZMKE AR FUKCMQ Y HDIM, PRQDARKCDE BEVMPDTE AR HEQ BMFEQ. CMIM CESR QUNMKRP, HE VRJRTECDMK RQ RQCEQE M EHCEKZE IUY NMCE EHTUPE. KM QR VRK EPSMHRQ JPEKARQ M NRÑURLMQ. OECDE RH RXTPRIM MCCDARKTEH, AMKAR QR OEHHE RH BURPTR IMUHTPDR Y QR EHZEK EHJUKEQ IDQRPESHRQ CMKQTPUCCDMKRQ OESDTEAEQ RK VRPE